### ISSUE TITLE
gobgp doesn't support VPNv6 over IPv4 (RFC 4798) ?

### ISSUE THREAD

**Issue by DKCopy** [osrg/gobgp#2113]
Version: gobgp 2.6.0
I use gobgp connect to Huawei switch (CE12800). when I enable  VPNv6 over IPv4 in Huawei switch,  the gobgpd trigger a panic!
So the gobgp maybe can't support VPNv6 over IPv4 (RFC 4798) ?
 
### Huawei switch bgp config
```
bgp 200
 peer 1.1.1.9 as-number 200
 peer 1.1.1.9 connect-interface 3.3.3.9
 #
 ipv4-family unicast
  peer 1.1.1.9 enable
 #
 ipv4-family vpnv4
  policy vpn-target
  peer 1.1.1.9 enable
 #
 ipv4-family vpn-instance vpna
  import-route direct
 #
 ipv6-family vpnv6
  policy vpn-target
  peer 1.1.1.9 enable
 #
 ipv6-family vpn-instance vpna
  import-route direct
#

```

### gobgp log
```
DEBU[0017] received update                               Key=3.3.3.9 Topic=Peer attributes="[{Origin: ?}  {Med: 0} {LocalPref: 100} {Extcomms: [65040:1]} {MpReach(l3vpn-ipv6-unicast): {Nexthop: 3.3.3.9, NLRIs: [3.3.3.9:11:2001:9:9:9::/64]}}]" nlri="[]" withdrawals="[]"
DEBU[0017] create Destination                            Nlri="3.3.3.9:11:2001:9:9:9::/64" Topic=Table
DEBU[0017] computeKnownBestPath knownPathList: 1         Topic=Table
DEBU[0017] From same AS, ignore.                         Data="{ 3.3.3.9:11:2001:9:9:9::/64 | src: { 3.3.3.9 | as: 200, id: 3.3.3.9 }, nh: 3.3.3.9 }" Key=3.3.3.9 Topic=Peer
DEBU[0017] received update                               Key=3.3.3.9 Topic=Peer attributes="[{MpUnreach(l3vpn-ipv6-unicast): End-of-Rib}]" nlri="[]" withdrawals="[]"
DEBU[0017] EOR received                                  AddressFamily=l3vpn-ipv6-unicast Key=3.3.3.9 Topic=Peer
DEBU[0017] send command to zebra                         Body="&{FRR_ZAPI6_INTERFACE_ADD [0xc00020fa00]}" Header="{10 254 6 0 FRR_ZAPI6_NEXTHOP_REGISTER}" Topic=Zebra
DEBU[0017] send command to zebra                         Body="&{FRR_ZAPI6_ROUTE_BGP 0 5 69 FRR_ZAPI5_SAFI_UNICAST {0 24 9.9.9.0} {0 0 <nil>} [{NEXTHOP_TYPE(0) 0 0 3.3.3.9 0 1 [17]}] 0 0 0 0 [0 0 0 0 0 0] FRR_ZAPI6_INTERFACE_ADD}" Header="{10 254 6 6 FRR_ZAPI6_ROUTE_ADD}" Topic=Zebra
DEBU[0017] send command to zebra                         Body="&{FRR_ZAPI6_INTERFACE_ADD [0xc00020fca0]}" Header="{10 254 6 6 FRR_ZAPI6_NEXTHOP_REGISTER}" Topic=Zebra
DEBU[0017] send command to zebra                         Body="&{FRR_ZAPI6_INTERFACE_ADD [0xc00020fd60]}" Header="{10 254 6 0 FRR_ZAPI6_NEXTHOP_REGISTER}" Topic=Zebra
panic: interface conversion: bgp.AddrPrefixInterface is *bgp.LabeledVPNIPv6AddrPrefix, not *bgp.LabeledVPNIPAddrPrefix

goroutine 24 [running]:
github.com/osrg/gobgp/pkg/server.addMessageLabelToIPRouteBody(0xc0001a3630, 0xc000000006, 0xc0000bf220, 0xc0000d9af2, 0xc0000d9ba0)
        /home/fujita/go/src/github.com/osrg/gobgp/pkg/server/zclient.go:124 +0x1c3
github.com/osrg/gobgp/pkg/server.newIPRouteBody(0xc0000d9da0, 0x1, 0x1, 0x6, 0xc0000bf220, 0xc0001ffb90, 0x1)
        /home/fujita/go/src/github.com/osrg/gobgp/pkg/server/zclient.go:169 +0x505
github.com/osrg/gobgp/pkg/server.(*zebraClient).loop(0xc0000bf220)
        /home/fujita/go/src/github.com/osrg/gobgp/pkg/server/zclient.go:437 +0xd3f
created by github.com/osrg/gobgp/pkg/server.newZebraClient
        /home/fujita/go/src/github.com/osrg/gobgp/pkg/server/zclient.go:527 +0x7e3


```

**Comment by fujita** [comment]
I guess that the zebra Integration can't handle this. Try disable the zebra feature.
@irino

**Comment by DKCopy** [comment]
> I guess that the zebra Integration can't handle this. Try disable the zebra feature.
> @irino

Linux 5.2 add support to routing ipv4 over ipv6 & ipv6 over ipv4! 
the gobgpd  maybe add this feature!

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d15662682db232da77136cd348f4c9df312ca6f9

**Comment by irino** [comment]
@DKCopy My older code makes panic. I fixed it. However Zebra does not support VPNv6 over IPv4 (https://github.com/FRRouting/frr/issues/4661). GoBGP depends on Zebra for installing route to kernel RIB, hence GoBGP cannot install VPNv6 routes to kernel RIB.
@fujita I found a bug for displaying vpnv6 using gobgp command, I also fix it in PR #2121.

**Comment by fujita** [comment]
@irino thanks for the comment. closed.
